core-tests/src/test/java/org/mule/runtime/core/internal/el/dataweave/DataWeaveExpressionLanguageAdaptorTestCase.java

```diff
@@ -550,6 +550,37 @@ public void bindingValueNotReferencedByDWParserCache() throws MuleException {
     }, "A hard reference is being mantained to the bindingValue."));
   }
 
+  @Test
+  public void handlesOptional() throws Exception {
+    TypedValue result =
+        expressionLanguage.evaluate("'hello' as String {class: 'java.util.Optional'}", testEvent(),
+                                    BindingContext.builder().build());
+    assertThat(result.getValue(), is(instanceOf(Optional.class)));
```

Comment:
not checking the value? maybe create an OptionalMatcher that checks the Class and the value